### PR TITLE
Add Kubeconfig download

### PR DESCRIPTION
**Pull request by dariadomagala**
**Description**

Changes proposed in this pull request:

- Add kubeconfig download in the context switcher and clusters list
- Add oidc data to the kubeconfig if token is missing
- Add UI tests

**Related issue(s)**

### CODE REVIEW

**Review by akucharska — COMMENTED** [review]
I tested it on cluster and it works well. I didn't have time to check code.

Although I encountered one random behavior. The whole app crushed when I had 2 clusters defined (and one of them was added from old kubeconfig). Unfortunately I wasn't able to replicate this so this might be unrelated.

**Comment by valentinvieriu** [comment]
I'm getting a strange error when I use enkode params
image

File: tests/tests/download-a-kubeconfig.spec.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/// <reference types='cypress' />
+
+const path = require('path');
+const downloadsFolder = Cypress.config('downloadsFolder');
+const kubeconfigPath = path.join(downloadsFolder, 'kubeconfig.yaml');
```
Comment (akucharska) [inline tests/tests/download-a-kubeconfig.spec.js:5]: Rename the variables to uppercase

**Review by akucharska — COMMENTED** [review]
Nice :)

One small comment below

**Comment by dariadomagala** [comment]
@valentinvieriu your problem is caused by the old init params (with the old schema) still saved in the session storage. Please remove session storage and try it out again.

**Review by akucharska — APPROVED** [review]

**Pull request merged**